File: leetcode/solutions/0383.RansomNote/canConstruct_test.go
package solution

import (
	"testing"
)

func Test_0383_RansomNote(t *testing.T) {
	t.Parallel()

	type in struct {
		arg1 string
		arg2 string
	}
	tests := []struct {
		in  in
		out bool
	}{
		{
			in{"a", "b"},
			false,
		},
		{
			in{"aa", "ab"},
			false,
		},
		{
			in{"aa", "aab"},
			true,
		},
		{
			in{"abc", "aab"},
			false,
		},
	}

	for _, tt := range tests {
		tt := tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			got := canConstruct(tt.in.arg1, tt.in.arg2)

			if got != tt.out {
				t.Errorf("got %#v want %#v", got, tt.out)
			}
		})
	}
}

File: leetcode/solutions/0383.RansomNote/canConstruct_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import canConstruct from "./canConstruct.ts";

Deno.test("0383. Ransom Note", () => {
  type test = [[string, string], boolean];
  const tests: test[] = [
    [["a", "b"], false],
    [["aa", "ab"], false],
    [["aa", "aab"], true],
    [["abc", "aab"], false],
  ];

  for (const [args, want] of tests) {
    assertEquals(canConstruct(...args), want);
  }
});
